src/Tasks/FileIO/VerifyFileHash.cs

```diff
@@ -54,25 +49,35 @@ public override bool Execute()
                 return false;
             }
 
+            if (!GetFileHash.TryParseHashEncoding(HashEncoding, out var encoding))
+            {
+                Log.LogErrorFromResources("FileHash.UnrecognizedHashEncoding", HashEncoding);
+                return false;
+            }
+
             byte[] hash = GetFileHash.ComputeHash(Algorithm, File);
-            if (!string.IsNullOrEmpty(Hash))
+            bool hashesMatch;
+            switch (encoding)
             {
-                var actualHash = ConversionUtilities.ConvertByteArrayToHex(hash);
-                if (!string.Equals(actualHash, Hash, StringComparison.OrdinalIgnoreCase))
-                {
-                    Log.LogErrorFromResources("VerifyFileHash.HashMismatch", File, Algorithm, Hash, actualHash);
-                    return false;
-                }
+                case Tasks.HashEncoding.Hex:
+                    var actualHash = ConversionUtilities.ConvertByteArrayToHex(hash);
+                    hashesMatch = string.Equals(actualHash, Hash, StringComparison.OrdinalIgnoreCase);
+                    break;
+                case Tasks.HashEncoding.Base64:
+                    byte[] expectedHash = Convert.FromBase64String(Hash);
+                    hashesMatch = expectedHash.SequenceEqual(hash);
+                    break;
+                default:
+                    throw new NotImplementedException();
             }
-            else
+
+            if (!hashesMatch)
             {
-                byte[] expectedHash = Convert.FromBase64String(HashBase64);
-                if (!expectedHash.SequenceEqual(hash))
-                {
-                    var actualHash = Convert.ToBase64String(hash);
-                    Log.LogErrorFromResources("VerifyFileHash.HashMismatch", File, Algorithm, HashBase64, actualHash);
-                    return false;
-                }
+                var actualHash = encoding == Tasks.HashEncoding.Hex
```

Comment:
This should probably use the same EncodeHash function from GetFileHash. Perhaps it could be an extension method of HashEncoding? It would mean that if another encoding method were added it would be less places to change.

Reply:
Updated.